psi4/src/psi4/libmints/mintshelper.cc

```diff
@@ -2424,6 +2433,106 @@ std::map<std::string, SharedMatrix> MintsHelper::metric_grad(std::map<std::strin
     return gradient_contributions;
 }
 
+SharedMatrix MintsHelper::embpot_grad(SharedMatrix D) {
+    // Computes skeleton (Hellman-Feynman like) derivatives for an arbitrary potential
+    int natom = molecule_->natom();
+    bool puream = basisset_->has_puream();
+
+    auto ret = std::make_shared<Matrix>("EMBPOT dervatives (i.e. Nx3)", natom, 3);
+    double **Gp = ret->pointer();
+    double **Dp = D->pointer();
+
+    FILE* input = fopen("EMBPOT", "r");
+    int npoints;
+    int statusvalue = fscanf(input, "%d", &npoints);
+    double x, y, z, w, v;
+    std::vector<double> xyz(npoints * 3);
+    std::vector<double> w_vec(npoints);
+    std::vector<double> v_vec(npoints);
+
+    // Pull out data
+    for (int k = 0; k < npoints; k++) {
+        statusvalue = fscanf(input, "%lf %lf %lf %lf %lf", &x, &y, &z, &w, &v);
```

Comment:
Should check the value of `statusvalue` to ensure the read in was successful.

Reply:
Perhaps something like:
```

if (statusvalue != 5) {
    throw PSIEXCEPTION("MintsHelper::embpot_grad error: EOF or wrong number of inputs read from EMBPOT, return=" + std::to_string(statusvalue)");
}
```